plugins/node/opentelemetry-instrumentation-aws-lambda/package.json

```diff
@@ -60,6 +59,7 @@
     "@opentelemetry/instrumentation": "^0.19.0",
     "@opentelemetry/resources": "^0.19.0",
     "@opentelemetry/semantic-conventions": "^0.19.0",
-    "@opentelemetry/tracing": "^0.19.0"
+    "@opentelemetry/tracing": "^0.19.0",
+    "@types/aws-lambda": "8.10.76"
```

Comment:
Usually we don't pin dependencies, only dev-dependencies

Reply:
I just moved them to different location, don't want to change them in  any other way

Reply:
I think we actually do pin all deps that aren't `@opentelemetry` deps usually iirc

Reply:
The only dependency I found is semver e.g. [here](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/45e875167e15f24003a2baf8b69263923b632e19/plugins/node/opentelemetry-instrumentation-dns/package.json#L67) which is not pinned.

We should be careful which version we use for type definitions. If npm is not able to de-duplicate them users end up in having two versions of e.g. @types/aws-lambda in the same project which usually not what someone wants.

I doubt that typescript is able to use version A for one file and version B for other files. And clearly types will differ between versions.

Reply:
We might want to think about using `*` if that is the case.

In this case, I think it's probably ok since we don't "leak" any of the types from the packages we wrap. If we use `@types/pg@1` and the user has `@types/pg@2`, ours will simply be installed in the deep node_modules directory at something like `node_modules/@opentelemetry/instrumentation-pg/node_modules/@types/pg` no?

Reply:
Hard to tell without trying.
Main question is what is typescript using - and remembering - when transpiling a file like this and the user has @types/pg installed in a significant different version then @opentelemetry/instrumentation-pg.
```typescript
import * from "pg"
import * from "@opentelemetry/instrumentation-pg"
```

I assume as long as the only types used are `protected init(): (InstrumentationNodeModuleDefinition<typeof pgTypes> | InstrumentationNodeModuleDefinition<typeof pgPoolTypes>)[];` in an protected method never used by end users it's don't care.

Once there are APIs using more types it may get problematic.

maybe we should somehow hide/remove the types from `InstrumentationAbstract#init` to avoid this at all?

Reply:
Guys you are aware that because we upgraded to ts 4 now all the packages doesn't contain the types and if you don't install the package by yrself it cannot be used. This change is just fixing to how it was, should be already merged and released. Then we can discuss about all other things but currently the packages are broken especially the meta package.

Reply:
@obecny is right. this simple program fails to compile because of missing types currently:

```ts
import type {} from '@opentelemetry/auto-instrumentations-node';
```

Reply:
A simple local test seems to show that typescript correctly differentiates the types. I manually installed `@types/pg` under `node_modules/@opentelemetry/instrumentation-pg/node_modules/@types/pg` and installed it also at `node_modules/@types/pg`. I made a breaking incompatible change to the types manually to the `node_modules/@types/pg`. My program still compiled successfuly and didn't break the checks for `node_modules/@opentelemetry/instrumentation-pg` so I think @Flarna's version concerns are OK.

Reply:
How would that be a different if you keep types in dev and compile using ts 3.x ? you would still break it exactly the same ?. The files will be in the same location if you either keep in dev with ts3 or move it to main packages and compile with ver. 4. You will have exactly the same pinned version of types right ?